programs/stake/src/stake_state.rs

```diff
@@ -560,7 +560,11 @@ impl Stake {
     ) -> (u128, u64) {
         // if there is no newer credits since observed, return no point
         if new_vote_state.credits() <= self.credits_observed {
-            return (0, 0);
```

Comment:
Well, returning `(_, 0)` was never a good idea to begin with.... Semantically, `credits_observed` shouldn't be `0` in all cases. Although, it had not been used up to #13680, I really tempted to fix this at #13325. In retrospect, I really wish I just did it right at the time.......

Reply:
lessons learnt: never violate invariants (= `credits_observed` can't be `0`) nevertheless at-the-time code was irrelevant of the violation.

it surely will bite future maintainers like this.

Reply:
Oh dang... so obvious in retrospect!  How'd I miss that :man_facepalming: